src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -35,155 +46,608 @@ def deserialize_bom_json(file_path: Path) -> dict:
 
     Returns
     -------
-    dict
-        The BOM content as a dictionary.
+    Bom
+        The CycloneDX Bom object.
 
     Raises
     ------
     CycloneDXParserError
         If the bom.json file cannot be located or deserialized.
-
     """
-    # TODO: use the official CycloneDX library to validate and deserialize BOM files
-    # once merged: https://github.com/CycloneDX/cyclonedx-python-lib/pull/290
     if not os.path.exists(file_path):
         raise CycloneDXParserError(f"Unable to locate any BOM files at: {str(file_path.parent)}.")
 
+    # We use the `cyclonedx-python-library` library for deserialization following the example here:
+    # https://cyclonedx-python-library.readthedocs.io/en/v7.3.4/examples.html
     with open(file_path, encoding="utf8") as file:
+        json_data = file.read()
+        if defaults.getboolean("dependency.resolver", "validate", fallback=True):
+            try:
+                schema_version = defaults.get("dependency.resolver", "schema", fallback="1.6")
+                my_json_validator = JsonStrictValidator(SchemaVersion.from_version(schema_version))
+            except ValueError as error:
+                raise CycloneDXParserError(f"Unable to find schema validator for {schema_version}: {error}") from error
+            try:
+                validation_errors = my_json_validator.validate_str(json_data)
+            except MissingOptionalDependencyException as error:
+                # Allow MissingOptionalDependencyException.
+                logger.debug("JSON-validation was skipped due to %s", error)
```

Comment:
I wonder what `MissingOptionalDependency` means?
From reading the documentation:
- https://cyclonedx-python-library.readthedocs.io/en/latest/autoapi/cyclonedx/exception/index.html#cyclonedx.exception.MissingOptionalDependencyException
- https://cyclonedx-python-library.readthedocs.io/en/latest/install.html#extras
It seems to me that this exception is only raised if we try to perform the json/xml validation without installing the extras alongside with `cyclonedx-python-lib`. 
However, looking at our `pyproject.toml`, I am not sure if we are installing the `validation` extras required for this feature.

Reply:
Hmm but if we are not installing the optional dependencies as part of Macaron dev environment, how can we still perform the json and xml validation (as seen from the unit tests) :thinking:

Reply:
I suspect parts of the validation requires the optional dependency. Anyway, I will add the `extra` but prefer to keep the exception handling following the example to be on the safe side.

Reply:
See commit 5a38400417f92ad803993c36d0b25508fcd72f83